src/coreclr/nativeaot/System.Private.TypeLoader/src/Internal/Runtime/TypeLoader/TypeBuilderState.cs

```diff
@@ -306,9 +306,9 @@ public ushort NumVTableSlots
 
         // Sentinel static to allow us to initialize _instanceLayout to something
         // and then detect that InstanceGCLayout should return null
-        private static LowLevelList<bool> s_emptyLayout = new LowLevelList<bool>();
+        internal static readonly bool[] s_emptyLayout = [];
```

Comment:
You should not need to cache in a static field - it is a deoptimization to cache it.